components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java

```diff
@@ -759,14 +759,16 @@ public static API addAPIWithGeneratedSwaggerDefinition(APIDTO apiDto, String oas
             apiToAdd.setAsyncApiDefinition(asyncApiDefinition);
         }
 
+        //adding the api
+        apiProvider.addAPI(apiToAdd);
+
```

Comment:
Remove this. see the reason below

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/10626